app/src/main/java/io/homeassistant/companion/android/vehicle/EntityGridVehicleScreen.kt

```diff
@@ -160,28 +161,40 @@ class EntityGridVehicleScreen(
             if (entity.isExecuting()) {
                 gridItem.setLoading(entity.isExecuting())
             } else {
-                gridItem
-                    .setOnClickListener {
-                        Log.i(TAG, "${entity.entityId} clicked")
-                        if (entity.domain in MainVehicleScreen.MAP_DOMAINS) {
-                            val attrs = entity.attributes as? Map<*, *>
-                            if (attrs != null) {
-                                val lat = attrs["latitude"] as? Double
-                                val lon = attrs["longitude"] as? Double
-                                if (lat != null && lon != null) {
-                                    val intent = Intent(
-                                        CarContext.ACTION_NAVIGATE,
-                                        Uri.parse("geo:$lat,$lon")
-                                    )
-                                    carContext.startCarApp(intent)
+                if (entity.domain !in MainVehicleScreen.NOT_ACTIONABLE_DOMAINS || canNavigate(entity)) {
+                    gridItem
+                        .setOnClickListener {
+                            Log.i(TAG, "${entity.entityId} clicked")
+                            when (entity.domain) {
+                                in MainVehicleScreen.MAP_DOMAINS -> {
+                                    val attrs = entity.attributes as? Map<*, *>
+                                    if (attrs != null) {
+                                        val lat = attrs["latitude"] as? Double
+                                        val lon = attrs["longitude"] as? Double
+                                        if (lat != null && lon != null) {
+                                            val intent = Intent(
+                                                CarContext.ACTION_NAVIGATE,
+                                                Uri.parse("geo:$lat,$lon")
+                                            )
+                                            carContext.startCarApp(intent)
+                                        }
+                                    }
+                                }
+
+                                in MainVehicleScreen.SUPPORTED_DOMAINS -> {
+                                    lifecycleScope.launch {
+                                        entity.onPressed(integrationRepository)
+                                    }
+                                }
+
+                                else -> {
+                                    // No op
```

Comment:
Should it be more obvious that there is no tap action for this item, for example by setting the item to be disabled?

Reply:
There is no disabled option for GridItem

Reply:
Odd, but I guess that really shows Google doesn't intend you to have non-actionable grid items.

Reply:
yea to be fair I have seen an app that adds an information only GridItem which was the EV map for the car sensor testing.

Reply:
Looks like Auto already handles some of this for us: no highlight when tapped (touch) / greyed out highlight when selected (rotary input).

Reply:
yea they recommend not setting a click listener in these places which is why I opted to skip adding it for these entities